src/ucp/core/ucp_context.c

```diff
@@ -745,6 +742,30 @@ static void ucp_resource_config_str(const ucp_config_t *config, char *buf,
     }
 }
 
+static void ucp_fill_sockaddr_aux_tls_config(ucp_context_h context,
+                                             const ucp_config_t *config)
+{
+    unsigned num_sockaddr_aux_tls, idx;
+    ucp_rsc_index_t tl_id;
+
+    context->config.sockaddr_aux_rscs.bitmap   = 0;
+    context->config.sockaddr_aux_rscs.num_rscs = config->sockaddr_aux_tls.count;
+
+    if (config->sockaddr_aux_tls.count != 0) {
+        /* Get the number of sockaddr auxiliary transports for the client-server flow */
+        num_sockaddr_aux_tls = config->sockaddr_aux_tls.count;
+
+        for (idx = 0; idx < num_sockaddr_aux_tls; ++idx) {
+            for (tl_id = 0; tl_id < context->num_tls; ++tl_id) {
```

Comment:
let's reuse ucp_str_array_search for now
when using aliases in next PR could also use ucp_tls_array_is_present

Reply:
@yosefe you mean like this?

```
for (idx = 0; idx < num_sockaddr_aux_tls; ++idx) {
        context->config.sockaddr_aux_rscs_bitmap |=
                        ucp_str_array_search(tl_names, count,
                                             config->sockaddr_aux_tls.aux_tls[idx],
                                             NULL);
    }
```

i don't have a list of tl_names to search in. only the list the user provided (with aliases), while the actual resources names are needed here.

Reply:
yes, like this

Reply:
but there is no resources list to search in

Reply:
config->sockaddr_aux_tls.aux_tls is the list

Reply:
this is the list which holds the tls i'm looking for. but where to search them in?

Reply:
search the list for context->tl_rscs[tl_id].tl_rsc.tl_name

Reply:
i need to search the given aux_tls (config->sockaddr_aux_tls.aux_tls) inside a list of resources - there is no such list, only names per rsc (context->tl_rscs[tl_id].tl_rsc.tl_name).
but ucp_str_array_search() requires a list to search in as the first param. which list is this?

Reply:
@yosefe in this case, ucp_str_array_search() will return a bitmap that matches the 
config->sockaddr_aux_tls.aux_tls list, to which i dont have access to in the wireup_ep code when comparing to the given dev_name.

Reply:
so don't use its return value as is but set the bit to UCS_BIT(tl_id) if the return value is nonzero